.github/workflows/ci.yml

```diff
@@ -0,0 +1,43 @@
+name: ci
+
+# Default to 'contents: read', which grants actions to read commits.
+#
+# If any permission is set, any permission not included in the list is
+# implicitly set to "none".
+#
+# see https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-24.04
+    timeout-minutes: 5
+
+    strategy:
+      matrix:
+        go: ["1.22.8", "1.23.2"]
```

Comment:
As we're not doing binary releases in this repository, these are only for CI. I think we can specify these without minor version instead;

```yaml
        go: ["1.22", "1.23"]
```


One thing we can _try_ (if it works) is if it's happy with the minimum version we specify in `go.mod`, and make the matrix

- "minimum version we claim is supported"
- "previous stable" (latest -1)
- "current stable"

```yaml
        go: ["1.13", "1.22", "1.23"]
```

If we want to reduce maintenance, we could even use the `oldstable`  and `stable` keywords for this; see https://github.com/actions/setup-go?tab=readme-ov-file#using-stableoldstable-aliases


```yaml
        go: ["1.13", "oldstable", "stable"]
```

Reply:
Good point on maintenance reduction. Ran this in fork (https://github.com/austinvazquez/go-events/actions/runs/11367549140) seems to work well.